modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/dynamic/data/mapping/util/DDMIndexerUtil.java

```diff
@@ -15,29 +15,24 @@
 package com.liferay.asset.list.internal.dynamic.data.mapping.util;
 
 import com.liferay.dynamic.data.mapping.util.DDMIndexer;
+import com.liferay.osgi.util.service.Snapshot;
 
 import java.util.Locale;
 
-import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Reference;
-
 /**
  * @author Pavel Savinov
  */
-@Component(service = {})
 public class DDMIndexerUtil {
 
 	public static String encodeName(
 		long ddmStructureId, String fieldReference, Locale locale) {
 
-		return _ddmIndexer.encodeName(ddmStructureId, fieldReference, locale);
-	}
+		DDMIndexer ddmIndexer = _ddmIndexerSnapshot.get();
 
-	@Reference(unbind = "-")
-	protected void se(DDMIndexer ddmIndexer) {
-		_ddmIndexer = ddmIndexer;
+		return ddmIndexer.encodeName(ddmStructureId, fieldReference, locale);
 	}
 
-	private static DDMIndexer _ddmIndexer;
+	private static final Snapshot<DDMIndexer> _ddmIndexerSnapshot =
```

Comment:
@JanisZhang This util is only used by AssetListAssetEntryProviderImpl which is a componet, could you just inline it?